### ISSUE TITLE
Mastodon / activitypub / fediverse DM to nostr

### ISSUE THREAD

**Issue by alltheseas** [nostrability/nostrability#98]
## client initiating action
Any nostr app 

## client receiving or observing action (or lack thereof)
Any mastodon app

## timestamp

https://github.com/damus-io/damus/issues/2489

## what happens 
Damus, and I suspect any Nostr app DMs are incompatible with mastodon DM's. This is not surfaced to the nostr user. There is potential for negative user experience. 

## question

How should app devs and designers handle this incompatibility?

**Comment by Sjors** [comment]
> How might nostr devs handle non-private DMs on mastodon?

I have a pull request for Mostr that proposes to simply respond with an auto-reply: https://gitlab.com/soapbox-pub/mostr/-/issues/91

That is for DM messages going from Fediverse to Nostr.

This seems like the best solution until Mostr actually implements DM. Sending message from Mastodon to Nostr would be very simple using nip-04. The Mastodon user already has no expectation of encryption, knows the site admins on both sides can read along, and the main Mastodon client clearly warns about that.

But on the nostr receive side it would be obvious for everyone to see who you're receiving DM's from. Unless you do something like gift wraps. There might be a nice solution there.

The other way around is much trickier to deal with for [reasons I explain here](https://gitlab.com/soapbox-pub/mostr/-/issues/91).

I'm worried much more about the downgraded experience of a Nostr user sending a DM to a Mastodon user (suddenly becomes plain text) then the other way around. But it seems weird to implement DM in only direction. This is why I'm inclined to just disable it altogether, until something like (interoperable) MLS saves the day.

**Comment by alexgleason** [comment]
All encrypted messages eventually get converted into plaintext. On Primal for example, the message is decrypted to be displayed on my screen. Therefore, it is always a concern that any Nostr app could then send your decrypted messages somewhere else.

The issue with the bridge is that it's guaranteed that a third-party (mostr) would decrypt the message first, before sending it to ANOTHER third-party (ActivityPub server), until it's eventually received by the recipient. IF we implemented DMs over the bridge.

I get that. It's something people should be aware of.

But I am less bothered by this idea than I was nearly 2 years ago, especially for NIP-04 which people already believe is unsalvageable. There is _always_ a chance the recipient, or some middleman, will leak it. It is the responsibility of the sender to make informed choices about what they share with whom over which channels.

Anyway, I said all of that but I'm still probably not going to implement it. I just said it to argue for no reason. However if someone did implement it, I would not be against it.

**Comment by alltheseas** [comment]
To me, nostr is about empowering the end user to make informee choice and making them aware of tradeoffs.

I support the spirit of @alexgleason suggestion of warning nostr users who are considering sending DMs to mastodon

> This user is bridged from an external network. Your message may not be seen, and it could be intercepted by third parties.

I suggest the above as a best practice for nostr DM apps who allow their users to interact with mastodon.

**Comment by alltheseas** [comment]
are y'all aware of any nostr apps adding that DMs to fediverse are plain text, and can be read by admins?

**Comment by alexgleason** [comment]
@alltheseas No, but no bridges today actually bridge DMs. DMs to Mostr users just go into the void. It doesn't get delievered to any servers on the Fediverse, and just hangs out on whatever relays the Nostr user submitted it to.
